### PR TITLE
 Implemented baseURL 

### PR DESCRIPTION
**Pull request by Robinfr**
The baseURL allows you to prefix the ID with a baseURL. This is useful
when the page has a `base` tag which in some browsers (e.g. Firefox)
will break the URLs when referencing IDs.

Note, I branched this off https://github.com/gilbarbara/react-inlinesvg/pull/85

### CODE REVIEW

**Comment by gilbarbara** [comment]
Can you show me an example of this error happening?

**Comment by Robinfr** [comment]
@gilbarbara sure. 

Open these in Firefox:
- [Without base tag](http://plnkr.co/edit/ykU211Qgx2HczNnM5LXY?p=preview)
- [With base tag](http://plnkr.co/edit/syDHyECUmo5aJyw7LzwI?p=preview)

The explanation can be found [here](https://stackoverflow.com/questions/18259032/using-base-tag-on-a-page-that-contains-svg-marker-elements-fails-to-render-marke/18265336#18265336).

**Comment by Robinfr** [comment]
Let me know if you need anything else.

**Comment by gilbarbara** [comment]
Sorry!

I couldn't stop any differences between the two links in the latest firefox.
Is it a legacy thing?

**Comment by gilbarbara** [comment]
There are some .orig files and a generated standalone file that aren't needed.
Please remove those

File: README.md [hunk 1]
```diff
@@ -83,6 +83,17 @@ A boolean that create unique IDs for each icon by hashing it. Default is `true`.
 **uniqueHash** `string`
 A string to use with `uniquifyIDs`. Default to a random 8 characters string `[A-Za-z0-9]`
 
+**baseURL** `string`
+A string to prefix each ID with a baseURL. Use this in case you are embedding the SVGs on a page that uses the `base` tag.
```
Comment (gilbarbara) [inline README.md]: Here it should say "An URL to prefix IDs in case you are using the `base` tag"

File: README.md [hunk 2]
```diff
@@ -83,6 +83,17 @@ A boolean that create unique IDs for each icon by hashing it. Default is `true`.
...
+
+Most common usecase:
```
Comment (gilbarbara) [inline README.md]: No example. This won't be used by most users, just the description is enough

File: test/index.spec.js [hunk 3]
```diff
@@ -168,6 +168,23 @@ describe('react-inlinesvg', () => {
       });
     });
 
+    it('should prefix the ids with the baseURL', done => {
+      const wrapper = setup({
+        src: '/test/__fixtures__/style.svg',
+        preloader: (<div className="loader">loading</div>),
+        baseURL: 'testing',
```
Comment (gilbarbara) [inline test/index.spec.js]: Please use a proper url.

**Comment by Robinfr** [comment]
@gilbarbara I tested in the latest Firefox and you're right it is indeed fixed there so it is a legacy thing. In that case everything before Firefox Quantum will have been affected.

I will try to resolve the issues tomorrow.

**Comment by gilbarbara** [comment]
I could see this behavior in Firefox 50, but simply adding `/` to the SVG properties didn't help. Does it need to be a full URL?

Anyway, I'm not sure it's the best approach to set a static baseURL prop since it will break in the development env with localhost that won't use the base tag.

I've never had issues related to this problem, are you having it in your production app? Can you share a link?

**Comment by Robinfr** [comment]
It needs to be the full path, thats why I was suggesting `document.location.pathname` as the default, as it works in both local development and in production.

Unfortunately I can't link to the application as it's work related. I have been using this fix for a while there already though with great succes. But if you have a better approach, I'd be glad to hear it.

Op 22 feb. 2018 5:52 p.m. schreef Gil Barbara <notifications@github.com>:

I could see this behavior in Firefox 50, but simply adding / to the SVG properties didn't help. Does it need to be a full URL?

Anyway, I'm not sure it's the best approach to set a static baseURL prop since it will break in the development env with localhost that won't use the base tag.

I've never had issues related to this problem, are you having it in your production app? Can you share a link?

**Comment by Robinfr** [comment]
I've applied the requested changes.

**Comment by gilbarbara** [comment]
There are still `.orig` files in this PR, please review it again...

So, you don't to set an URL to handle the `base` tag but a pathname? Then the prop name and description are misleading.

**Comment by Robinfr** [comment]
@gilbarbara a pathname is a valid URL, it's a relative URL. You can also put an absolute URL, it really doesn't matter. I will remove the .orig files.

**Comment by Robinfr** [comment]
@gilbarbara .orig files removed.

**Comment by gilbarbara** [comment]
Dude, seriously?

I'm concerned about all users of this module, including tons of newbies which I don't want to discuss semantics, so it's always better to be clear on documentation.

Anyway, I'll merge this soon and finish the documentation.
Thanks

**Pull request merged**